src/services/passportStrategies/passportFacebookStrategy.ts

```diff
@@ -0,0 +1,40 @@
+import { Strategy as FacebookStrategy } from "passport-facebook";
+import type { VerifyFunction } from "passport-facebook";
+import { generateCrypto } from "../generateCrypto";
+import { findUserBy } from "../databaseServices/findUserBy";
+import { createFacebookUser } from "../databaseServices/createFacebookUser";
+import bcrypt from "bcrypt";
+import { SALT_ROUNDS } from "../../constants/bcrypt";
+
+const strategyOptions = {
+  clientID: <string>process.env.FACEBOOK_ID,
+  clientSecret: <string>process.env.FACEBOOK_SECRET,
+  callbackURL: <string>process.env.FACEBOOK_CALLBACK_URL,
+  profileFields: ["displayName", "emails"],
+};
+
+const verifyCallback: VerifyFunction = async (
+  _accessToken,
+  _refreshToken,
+  profile,
+  done
+) => {
+  if (profile?.emails) {
+    const username = profile.displayName;
+    const email = profile?.emails[0].value;
+    const user = await findUserBy({ email });
+    if (user) return done(null, user);
+    const crypto = await generateCrypto();
```

Comment:
`generateCrypto` returns an `authCode` yet it's used to generate the initial user's password